Executables/Game/World/Entities/MonsterEntity.cs

```diff
@@ -31,18 +31,26 @@ public class MonsterEntity : Entity
 
         public MonsterData Proto { get { return _proto; } }
 
-        public MonsterGroup Group { get; set; }
+        public MonsterGroup? Group { get; set; }
 
         private readonly MonsterData _proto;
-        private IBehaviour _behaviour;
+        private IBehaviour? _behaviour;
         private bool _behaviourInitialized;
         private double _deadTime = 5000;
 
         public MonsterEntity(IMonsterManager monsterManager, IAnimationManager animationManager, IWorld world, ILogger logger, uint id, int x, int y, float rotation = 0)
             : base(animationManager, world.GenerateVid())
         {
+            var proto = monsterManager.GetMonster(id);
+
+            if (proto is null)
+            {
+                // todo handle better
```

Comment:
Since this should not occur I think this is the best solution. It is a real issue if the ID cannot be matched to a known monster

Reply:
I think this is okay for now. It is possible that monsters cannot be found in the monster database (mob_proto). A spawn configuration may reference a non existing monster ID. However this will throw an exception and crash the whole server. Monsters are usually spawned on server (/map) startup so this should be fine for now.

Reply:
Agreed